src/quote/getQuoteAndBuyCoverInputs.ts

```diff
@@ -0,0 +1,161 @@
+import axios, { AxiosError } from 'axios';
+import dotenv from 'dotenv';
+
+import { calculatePremiumWithCommissionAndSlippage } from '../buyCover/calculatePremiumWithCommissionAndSlippage';
+import {
+  CoverAsset,
+  CoverId,
+  DEFAULT_COMMISSION_RATIO,
+  DEFAULT_SLIPPAGE,
+  MINIMUM_COVER_PERIOD,
+  NEXUS_MUTUAL_DAO_TREASURY_ADDRESS,
+  TARGET_PRICE_DENOMINATOR,
+} from '../constants/buyCover';
+import {
+  Address,
+  CoverRouterQuoteResponse,
+  ErrorApiResponse,
+  GetQuoteApiResponse,
+  GetQuoteResponse,
+  IntString,
+  Integer,
+  PoolCapacity,
+} from '../types';
+
+dotenv.config();
+
+type CoverRouterQuoteParams = {
+  productId: Integer;
+  amount: IntString;
+  period: Integer;
+  coverAsset: CoverAsset;
+};
+
+async function getQuoteAndBuyCoverInputs(
+  productId: Integer,
+  coverAmount: IntString,
+  coverPeriod: Integer, // days
+  coverAsset: CoverAsset,
+  coverBuyerAddress: Address,
+  paymentAsset: CoverAsset = coverAsset,
+  slippage: number = DEFAULT_SLIPPAGE, // 0.1%
+  ipfsData: string = '', // IPFS base32 hash value
+): Promise<GetQuoteApiResponse | ErrorApiResponse> {
+  let result: GetQuoteResponse | undefined = undefined;
+
+  if (!process.env.COVER_ROUTER_URL) {
+    return { result, error: { message: 'Missing COVER_ROUTER_URL env var', data: {} } };
+  }
+
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return { result, error: { message: 'Invalid productId: must be a positive integer', data: {} } };
+  }
+
+  if (!/^\d+$/.test(coverAmount) || parseInt(coverAmount, 10) <= 0) {
+    return { result, error: { message: 'Invalid coverAmount: must be a positive integer string', data: {} } };
```

Comment:
On FE, we allow non-integer values for amount as well. Not sure if this validation is a requirement for Etherfi integration only, just wanted to flag it.

Reply:
we should allow non-integer values as well.

Reply:
fixed

Reply:
as discussed in with @mixplore in slack, since the SDK is aimed for use with devs all inputs and outputs will be in the smallest currency (i.e. wei)

hence reverting this to allow only integer strings, non-integer values will be invalid inputs.